Shared/Samples/Orbit camera around object/OrbitCameraAroundObjectView.Model.swift

```diff
@@ -0,0 +1,176 @@
+                
+                let pitchDelta = pitchDelta(for: 45)
+                cameraController.autoPitchIsEnabled = false
+                
+                await cameraController.moveCamera(
+                    distanceDelta: 50 - cameraController.cameraDistance,
+                    headingDelta: -cameraController.cameraHeadingOffset,
+                    pitchDelta: pitchDelta,
+                    duration: 1
+                )
+                _ = try await cameraController.setTargetOffsets(x: 0, y: 0, z: 0, duration: 1)
+            }
+            
+            // Restrict the camera's pitch so it doesn't collide with the ground.
+            cameraController.minCameraPitchOffset = 10
+            cameraController.maxCameraPitchOffset = 100
+            
+            cameraController.minCameraDistance = 10
+            cameraController.cameraDistanceIsInteractive = true
+        }
+        
+        /// Moves the view of the camera controller to the cockpit of the plane.
+        func moveToCockpit() async throws {
+            cameraController.cameraDistanceIsInteractive = false
+            cameraController.minCameraDistance = 0.1
+            
+            // Unlock the camera pitch for the rotation animation.
+            cameraController.minCameraPitchOffset = -180
+            cameraController.maxCameraPitchOffset = 180
+            
+            // Animate the camera to the cockpit, facing forward (0° heading),
+            // and aligned with the horizon (90° pitch).
+            _ = try await cameraController.setTargetOffsets(x: 0, y: -1.4, z: 1.3, duration: 1)
+            await cameraController.moveCamera(
+                distanceDelta: 0.1 - cameraController.cameraDistance,
+                headingDelta: -cameraController.cameraHeadingOffset,
+                pitchDelta: pitchDelta(for: 90),
+                duration: 1
+            )
+            
+            // Lock the camera pitch when the animation finishes.
+            cameraController.minCameraPitchOffset = 90
+            cameraController.maxCameraPitchOffset = 90
+            cameraController.autoPitchIsEnabled = true
+        }
+        
+        /// The camera pitch delta for a given angle and the current plane pitch.
+        /// - Parameter angle: The angle in degrees.
+        /// - Returns: The change in pitch.
+        private func pitchDelta(for angle: Double) -> Double {
+            let planePitch = planeGraphic.attributes["PITCH"] as? Double ?? 0
+            let cameraPitchOffset = cameraController.cameraPitchOffset
+            
+            if cameraController.autoPitchIsEnabled {
+                return angle - cameraPitchOffset
+            } else {
+                return angle + planePitch - cameraPitchOffset
+            }
+        }
+    }
+}
+
+private extension OrbitGeoElementCameraController {
+    /// The target focus point relative to the target object.
+    var targetOffsetPoint: Point {
```

Comment:
Since this point is only used in one place, maybe can be changed to sth like

```swift
var targetOffsetIsZero: Bool { [targetOffsetX, targetOffsetY, targetOffsetZ].allSatisfy(\.isZero) }
```